### ISSUE TITLE
Sensor Executor Adds Incorrect Git Artifact Input to Trigger

### ISSUE THREAD

**Issue by Syps** [argoproj/argo-events#354]
**Describe the bug**
***TL;DR***
Git input artifact does not work for workflow that is triggered by sensor whose event source is a git event.

***More details***
If you create a test workflow whose first step uses a git input artifact, it will work as exected when executed via `argo submit`; however, if you use the same workflow as a trigger in argo events for a sensor whose event source is a git event, the sensor executor almost seems to be overriding the trigger's git input with the git data from the event source.

**To Reproduce**

For example, take the following workflow:

_workflow.yaml_
```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: hello-world-
spec:
  entrypoint: whalesay
  templates:
  - name: whalesay
    inputs:
      artifacts:
      - name: pipeline
        path: /home/my_repo
        git:
          repo: https://my.origin/mygroup/myproject.git
          revision: my_branch
    container:
      image: docker/whalesay:latest
      command: [cowsay]
      args: ["hello world"]
      resources:
          limits:
            cpu: 1000m
            memory: 1Gi
```

Running this via `argo submit workflow.yaml` will succeed as expected.

However, now take the following manifest for an Argo Event configuration:

```
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: gitlab-event-source
  labels:
    # do not remove
    argo-events-event-source-version: v0.10
data:
  example: |-
    # id of the project
    projectId: "0"
    # Github will send events to following port and endpoint
    hook:
      # endpoint to listen to events on
      endpoint: "/merge-requests-events"
      # port to run internal HTTP server on
      port: "80"
      # url the gateway will use to register at Github.
      # This url must be reachable from outside the cluster.
      # The gateway pod is backed by the service defined in the gateway spec. So get the URL for that service Github can reach to.
      url: "http://gitlab-gateway-svc.website.com"
    event: "MergeRequestsEvents"
    accessToken:
      key: token
      name: gitlab-secret
    enableSSLVerification: false
    gitlabBaseUrl: "https://my.origin"
---
apiVersion: argoproj.io/v1alpha1
kind: Gateway
metadata:
  name: gitlab-gateway
  labels:
    gateways.argoproj.io/gateway-controller-instanceid: argo-events
    argo-events-gateway-version: v0.10
    expr.app: gitlab-gateway
spec:
  type: "gitlab"
  eventSource: "gitlab-event-source"
  processorPort: "9330"
  eventProtocol:
    type: "HTTP"
    http:
      port: "9300"
  template:
    metadata:
      name: "gitlab-gateway"
      labels:
        gateway-name: "gitlab-gateway"
    spec:
      containers:
        - name: "gateway-client"
          image: "argoproj/gateway-client"
          imagePullPolicy: "Always"
          command: ["/bin/gateway-client"]
          resources:
            limits:
              cpu: 100m
              memory: 500Mi
            requests:
              cpu: 50m
              memory: "104857600"
        - name: "gitlab-events"
          image: "argoproj/gitlab-gateway"
          imagePullPolicy: "Always"
          command: ["/bin/gitlab-gateway"]
          resources:
            limits:
              cpu: 100m
              memory: 500Mi
            requests:
              cpu: 50m
              memory: "104857600"
      serviceAccountName: "argo-events-sa"
  service:
    metadata:
      name: gitlab-gateway-svc
    spec:
      selector:
        gateway-name: "gitlab-gateway"
      ports:
        - port: 80
          targetPort: 9330
      clusterIP: None
      type: ClusterIP
  watchers:
    sensors:
      - name: "gitlab-sensor"
---
apiVersion: argoproj.io/v1alpha1
kind: Sensor
metadata:
  name: gitlab-sensor
  labels:
    sensors.argoproj.io/sensor-controller-instanceid: argo-events
    argo-events-sensor-version: v0.10
spec:
  template:
    spec:
      containers:
        - name: "sensor"
          image: "argoproj/sensor"
          imagePullPolicy: Always
          resources:
            limits:
              cpu: 100m
              memory: 500Mi
            requests:
              cpu: 50m
              memory: "104857600"
      serviceAccountName: argo-events-sa
  eventProtocol:
    type: "HTTP"
    http:
      port: "9300"
  dependencies:
    - name: "gitlab-gateway:example"
  triggers:
    - template:
        name: my_workflow
        group: argoproj.io
        version: v1alpha1
        kind: Workflow
        source:
          inline: |
            apiVersion: argoproj.io/v1alpha1
            kind: Workflow
            metadata:
              generateName: my-workflow-
            spec:
              entrypoint: main
              arguments:
                parameters:
                - name: git_url
                  value: https://my.origin/mygroup/myproject.git
                - name: git_branch
                  value: my_branch
              serviceAccountName: "argo-runner-sa"
              templates:
              - name: main
                steps:
                - - name: my-step
                    template: my-template

              - name: my-template
                container:
                  image: docker/whalesay:latest
                  command: [cowsay]
                  args: ["hello world"]
                  resources:
                    requests:
                      cpu: 1000m
                      memory: 1Gi
                    limits:
                       cpu: 1000m
                       memory: 1Gi
                inputs:
                  artifacts:
                  - name: pipeline
                    path: /home/myproject
                    git:
                      repo: "{{workflow.parameters.git_url}}"
                      revision: "{{workflow.parameters.git_branch}}"
      resourceParameters:
        - src:
            event: "gitlab-gateway:example"
          dest: spec.arguments.parameters.0.value
```

Apply this manifest to kubernetes namespace, and trigger the event. The workflow is executed, but the init container will fail to pull the git repository.

**Expected behavior**
In both cases, the workflow execution should behave the same (they should _both_ succeed)

**Additional context**
I started debugging this locally and I see argoexec (in argo workflows) using different git artifacts in each of the above cases. I built workflows locally and placed some log statements where the `gitClone` is happening in argoexec (specifically, I place the following at https://github.com/argoproj/argo/blob/e7e50af6e56b1eeddccc82a2dbc8b421d1a63942/workflow/artifacts/git/git.go#L90):

```
	mygit := inputArtifact.Git
	fmt.Println(*mygit)
```

Here's what I see from the print statement when running the workflow via `argo submit`:

```
{https://my.origin/mygroup/myproject.git my_branch <nil> [] nil nil nil true}
```

And here's what I see when the workflow is executed by the trigger in argo-events:

```
{{"object_kind":"push",
"event_name":"push",
"before":"",
"after":"",
"ref":"refs/heads/my_branch",
"checkout_sha":"",
"message":null,
"user_id":"0",
"user_name":"blablabla",
"user_username":"blabla1",
"user_email":"",
"user_avatar":"",
"project_id":0,
"project":{"id":0,"name":"myproject","description":"",
"web_url":"","avatar_url":"","git_ssh_url":"",
"git_http_url":"",
"namespace":"mynamespace",
"visibility_level":20,"path_with_namespace":"",
"default_branch":"master",
"ci_config_path":null,"homepage":"
","url":"g","ssh_url":"","http_url":""},
"commits":
[{"id":","message":"Update new.file","timestamp":"2019-09-15T20:44:41Z",
"url":"","author":{"name":"","email":""},"added":[],
"modified":["new.file"],"removed":[]}],
"total_commits_count":1,"push_options":{},
"repository":{"name":"my_proj","url":"","description":"",
"homepage":"","git_http_url":"","git_ssh_url":"",
"visibility_level":20}} my_branch <nil> [] nil nil nil true}
```

_(I've replaced/removed most of the actual values here)_

The above output makes me think that the sensor is adding incorrect parameters to the triggers git artifact input prior to creating the workflow resource.

**Comment by VaibhavPage** [comment]
The behavior you are seeing is expected. It is dues to the fact that the `path` is not specified in `resourceParameters src`. https://argoproj.github.io/argo-events/parameterization/#parameter-structure

You may want to define your `resourceParameters` as follows,
     resourceParameters:
        - src:
            event: "gitlab-gateway:example"
            path: "project.web_url" // the project.web_url means, `web_url `key within `project` key within event payload object. To read more on how to pick specific subset of json object - https://github.com/tidwall/gjson#path-syntax
            value: "https://my.origin/mygroup/myproject.git" // this would be default value if `project.web_url` doesnt exist in event payload.
          dest: spec.arguments.parameters.0.value

**Comment by Syps** [comment]
you're right, thank you!
